### ISSUE TITLE
Email maintainers about available money

### ISSUE THREAD

**Issue by nobodxbodon** [gratipay/inside.gratipay.com#969]
Reticketing from https://github.com/gratipay/inside.gratipay.com/issues/964#issuecomment-271064618

Original suggestion from a potential company giver:
>I'd gladly contribute to a project on Gratipay, but I'd like to contribute to a project that we actually use and benefit from at Loopify.
>
>Specifically, here's a list of npm modules I'm grateful for:
>- async
>- babel-* (also css-loader, file-loader, saas-loader, style-loader, url-loader, etc)
>- iframe-resizer
>- mongoose
>- nodemon
>- react
>- react-helmet
>- react-modal
>- react-redux
>- react-router
>- react-router-redux
>- redux
>- redux-thunk
>- webpack
>
>I'd suggest you hit up all these project maintainers and say "I've got a startup founder who wants to pay for your project if you sign up for Gratipay". Then let me know! :)

The channels of contacting may include email or other social accounts. The process will get direct feedback for npm integration project, and also find the proper means and words to contact project maintainers, which can be reused in future auto-notifying.

**Comment by nobodxbodon** [comment]
Started to get contacts. Some findings:

>- async

There are 4 admin [listed on npm](https://www.npmjs.com/package/async/access)
And of course much more [contributors](https://libraries.io/github/caolan/async/contributors)
I suppose we'll use the admin list.


>- react

It seems a [facebook project](https://github.com/facebook/react). I suppose the wording should be different from that used for individual project maintainers? It's still an issue how to automatically check whether a project is maintained by a company or individual.

**Comment by whit537** [comment]
Add to "Integrate npm" project?

**Comment by whit537** [comment]
[slack](http://gratipay.slackarchive.io/gratipay/-/1484238750.002521/1484251287.002637/1484251132002630/)

**Comment by nobodxbodon** [comment]
Though I get the responses from @whit537 that's referenced in above slack archive already, please allow me to make another (hopefully the last, either way :)) try to highlight the necessity in my view to reach out to the project owners.

As @arasmussen said "let me know", I believe the result (how many of the project owners are willing to accept giving from another company) would actually in turn impact how his company participates in the project. Besides, I wonder if he's aware that some projects are company project already (react by facebook, for example), which I also think would influence his decision.

Maybe I'm wrong about this, or are there some obvious answers that I don't realize?

**Comment by arasmussen** [comment]
Happy to add some context here. I've got plenty of thoughts.

1. I don't want to pay Facebook. They've got plenty of money and open source is very well funded there. Any money I could contribute would be a rounding error. Nothing against FB, I used to work there, just don't want to give them money.

2. We're bootstrapped and currently losing money, so it's hard for me to justify paying money to anyone right now. When/if that situation changes, I'd definitely love to contribute to any projects that helped us succeed.

3. I don't want to distribute to all of my npm modules evenly. Some of them I don't use, some of them are built by FB, some of them are wayyyyy more useful than others, some aren't actively maintained. I want to fund the people who are doing open source work for the greater good of software and open source, who work on great projects used by influential companies, who we might lose because they need to make a living...

Some npm modules I love and would consider funding, assuming the above conditions are met (we're profitable, the project isn't already well funded): async, babel, linkifyjs, mocha, moment, nodemon, react, react-helmet, react-redux, react-router, react-router-redux, redux, redux-connect, redux-thunk, webpack.

Hope that's helpful, cheers!

**Comment by arasmussen** [comment]
> So you'd rather choose a la carte rather than get a bundle based on dependencies? That seems easier to implement.

Yeah or maybe a page where I can copy + paste my package.json in and then check the boxes for the projects I'd like to contribute to.

> Another idea that comes from "isn't already well funded": what if you could put a ceiling on contributions? E.g., if project x already receives $1000 a week, put my money elsewhere.

Yeah $1000 a week is solid for one person, not for a team of people. Really depends on the project. I'm less worried about this, more worried about not giving my money to Facebook or Facebook engineers who are already making six figures doing what they love. It's more about "don't fund projects well-funded by big companies".

**Comment by mattbk** [comment]
$1000 was a random number. That's fine. I like the box-checking, but would a "fill in an amount" list work better for what you're describing? This is one of those choices we could build; divide money equally among X projects, or divide it up manually.

**Comment by nobodxbodon** [comment]
> I'm less worried about this, more worried about not giving my money to Facebook or Facebook engineers who are already making six figures doing what they love.

As it's critical, we'd better try to make sure of that. I don't see other ways better than reaching out to the project maintainers to check if they'd like to receive it. I suppose the professionals like full-time engineers would either reject or make that clear in response?

**Comment by nobodxbodon** [comment]
And thanks @arasmussen a lot for your precious feedback!

**Comment by mattbk** [comment]
> I don't see other ways better than reaching out to the project maintainers to check if they'd like to receive it.

A key part of this infrastructure is that we're not collecting money on behalf of those projects until they sign up to receive it, only pledges. I think this is reasonable if we make it very clear which projects aren't signed up yet _and_ give those maintainers a simple way to opt-out and be removed entirely.

**Comment by nobodxbodon** [comment]
@mattbk I'm not sure I get how it's going to work:

1. a company signs up, ready to pay for certain projects (do we ask how much in their mind, either total or for each project?)
2. we wait(?) for the project owners to sign up, without reaching out to them proactively? <-- **the problematic one**
3. after the project owner signs up, the company can give (not necessarily the amount in the beginning, as their budget can change, fewer projects to pay, etc)

**Comment by mattbk** [comment]
I see what you're saying. :+1: I thought this was in conjunction with things like https://github.com/gratipay/gratipay.com/issues/4148 and https://github.com/gratipay/gratipay.com/pull/4135. That's where I was coming from pledging.

I completely agree with reaching out to potential project owners once we have an idea of who wants to give to them. Ideally, the first few companies will be a little more patient and we can run through this manually, and then more projects will join as they realize there is potential funding involved.

**Comment by whit537** [comment]
I'm removing this from the `Integrate npm` project after all, after working on https://github.com/gratipay/inside.gratipay.com/issues/987. Let's discuss where this issue fits under there.

**Comment by whit537** [comment]
Added to a new project board for the epic.

**Comment by whit537** [comment]
I've privately emailed someone who maintains lots of npm projects to ask for a conversation. Will report back ...

**Comment by whit537** [comment]
I thought I dug up at some point a list of the top 10 npm maintainers by number of packages. We could reach out to more folks based on that list but I didn't want to wait for that before reaching out to this one person I *know* is on the list and who came to mind and seemed like a good person to reach out to based on past history.

**Comment by whit537** [comment]
No word back yet. Moving back to take-off in light of https://github.com/gratipay/inside.gratipay.com/issues/987#issuecomment-274151963.

**Comment by whit537** [comment]
Heard back! Scheduling a call ...

**Comment by nobodxbodon** [comment]
@whit537 how's the call going? Guess not done yet?

**Comment by whit537** [comment]
Done: https://www.youtube.com/watch?v=B-kC1SEx31g.

Needs to be summarized. Up for that?

**Comment by whit537** [comment]
On Doug's recommendation, I've emailed [Jonathan Ong](https://github.com/jonathanong).

**Comment by whit537** [comment]
The tl;dr of my call with Doug is that he thought the integration w/ npm is interesting enough that he would be [willing](https://www.youtube.com/watch?v=B-kC1SEx31g#t=46m36s) to sign up some of his projects ([express](https://gratipay.com/on/npm/express/), [mysql](https://gratipay.com/on/npm/mysql/)) once we are ready to go. 💃

One key [insight](https://www.youtube.com/watch?v=B-kC1SEx31g#t=42m40s) is that on [the fairness question](https://www.youtube.com/watch?v=B-kC1SEx31g#t=38m15s) ([cf.](https://github.com/gratipay/gratipay.com/pull/4135#issuecomment-263386065)), it might actually be better *not* to distribute to the full transitive dependency graph, but just to the top level. That way, that first tier holds the reigns in distributing to the next level down.

**Comment by whit537** [comment]
Jonathan isn't interested in talking, as he's not doing as much with open source these days.

**Comment by mattbk** [comment]
> That way, that first tier holds the reigns in distributing to the next level down.

I can see that, as each successive level probably(?) knows more about its direct dependencies than the ones further down.

**Comment by nobodxbodon** [comment]
>> That way, that first tier holds the reigns in distributing to the next level down.

> I can see that, as each successive level probably(?) knows more about its direct dependencies than the ones further down.

+1 to the distribution of responsibilities. Though it reminds me of [trickle-down effects](https://en.wikipedia.org/wiki/Trickle-down_economics), IMO it's still a proper mechanism for at least the early stage.

BTW trickle-down here may not be as serious as in the real world, as there can be lots of 'leaf' projects giving to one single 'root' project.

**Comment by whit537** [comment]
[Top 100 npm Users By Downloads](https://gist.github.com/bcoe/dcc961b869bbf6685002)

**Comment by whit537** [comment]
Relevant from https://github.com/sustainers/website/issues/56#issuecomment-298692559:

> Not so secret list of the 1,000 most depended upon projects tracked by Libraries.io:
>
> https://gist.github.com/andrew/872df3b78b9ec3e7dbe597fb5a202121

**Comment by whit537** [comment]
Dropped this down a notich in scope (~= bumped this up a notch in priority) from Epic to Project.

**Comment by whit537** [comment]
Let's repurpose this to be about emailing maintainers when someone pledges to them.

**Comment by whit537** [comment]
I.e., automated outreach.

**Comment by whit537** [comment]
@oakes in [slack](https://gratipay.slackarchive.io/gratipay/page-100/ts-1495724983367112):

> you could do what Brave does, and send an email to project maintainers when their donations have reached a certain amount (i think Brave does it at $100)

**Comment by whit537** [comment]
Repurposing slightly in light of https://github.com/gratipay/inside.gratipay.com/issues/1160. Now this is about emailing maintainers once we decide how we're going to distribute funds raised through #BackTheStack.
